src/ansys/dpf/composites/_indexer.py

```diff
@@ -66,6 +66,35 @@ def by_id(self, entity_id: int) -> Optional[NDArray[np.int64]]:
         """Get index by id."""
 
 
+def _has_data_pointer(field: PropertyField | Field) -> bool:
```

Comment:
The `X | Y` syntax for types is a Python 3.10 feature. It's probably easiest to just use `Union[X, Y]` for now. Alternatively, we can first drop 3.9 support.